proxy/logging/LogObject.cc

```diff
@@ -988,7 +988,6 @@ LogObjectManager::_solve_filename_conflicts(LogObject *log_object, int maxConfli
             roll_file = false;
           } else {
             if (S_ISFIFO(s.st_mode)) {
-              unlink(filename);
```

Comment:
Testing this out locally, removing the unlink seems to get the behavior that addresses the concerns @ema raised.

However, do we have to worry about the comment at line 974:
https://github.com/apache/trafficserver/pull/6017/files#diff-75bc20fd8082b64381dec2cf402275feR974

It seems to indicate that the unlink is intentional so that the metadata is correct. Maybe the comment is incorrect though?

Reply:
what does the meta data files say about the fifo in your testing?

Reply:
With the changes from this patch (i.e., unlink is happening) on a production box, here's the contents of the pipe's meta file:

```
-bash-4.2$ ls -la squid.log .squid.log.meta 
-rw-r--r-- 1 nobody nobody 439705513 Oct 25 15:17 squid.log
-rw-r--r-- 1 nobody nobody        66 Oct 25 15:15 .squid.log.meta
-bash-4.2$ date
Fri Oct 25 15:17:57 UTC 2019
-bash-4.2$ cat .squid.log.meta 
creation_time = 1572016504
object_signature = 2623008075644961995
```

Reply:
Reading through the code, I see that the meta file is only used for log rolling which is not applicable to pipes. We don't even instantiate a BaseLogFile (which creates and maintains the meta file) for pipes:

```
LogFile::LogFile(const char *name, const char *header, LogFileFormat format, uint64_t signature, size_t ascii_buffer_size,
                 size_t max_line_size)
  : m_file_format(format),
    m_name(ats_strdup(name)),
    m_header(ats_strdup(header)),
    m_signature(signature),
    m_max_line_size(max_line_size)
{
  if (m_file_format != LOG_FILE_PIPE) {    <<<<<<<< Note created for pipes.
    m_log = new BaseLogFile(name, m_signature);
    m_log->set_hostname(Machine::instance()->hostname);
  } else {
    m_log = nullptr;
  }
```

In any case, the comment about unlinking the pipe to maintain the metadata for it seems to be a mistaken comment (assuming by metadata the author meant the meta file).

@ericcarlschwartz : I'll create a new PR with the close() call and manage the PR from there.

Thanks!
Brian

Reply:
I created a PR to continue this work here:

https://git.ouroath.com/Edge/trafficserver/pull/111